radix-transactions/src/model/preparation/traits.rs

```diff
@@ -199,17 +203,28 @@ macro_rules! transaction_payload_v2 {
     };
 }
 
-pub(crate) use transaction_payload_v2;
+pub(crate) use define_transaction_payload;
 
 pub trait RawTransactionPayload: AsRef<[u8]> + From<Vec<u8>> + Into<Vec<u8>> {
+    const KIND: TransactionPayloadKind;
+
     fn as_slice(&self) -> &[u8] {
         self.as_ref()
     }
 }
 
+pub trait ValidatedTransactionPayload: IntoExecutable {}
+
+#[derive(Debug, Copy, Clone)]
+pub enum TransactionPayloadKind {
+    CompleteUserTransaction,
+    LedgerTransaction,
+    Other,
```

Comment:
it seems a bit safer if we explicitly listed all kinds here?

Reply:
Hehe I wasn't sure if I missed any other usages or it was really just the length check :) I like both options